### ISSUE TITLE
Moving Frames and Keys at the Same Time

### ISSUE THREAD

**Issue by turtletooth** [opentoonz/opentoonz#1665]
Every one in a hundred times I am able to do some magic with key modifiers and move a frame and its key together.  There is also a preference to move frames and cell data together, but that only seems to work when moving from column to column, not within the same column.  

@manongjohn Am I missing anything, or is this a bug?

**Comment by manongjohn** [comment]
I am a familiar with the preference that allows frames and cell data to move together between columns, but not sure about within the same column.

When I was testing the flip, I checked that against an earlier release and didn't see that behavior within the same column before.  Only when the preference was set and only when moving to a new column.

If moving cell data within the same column regardless of preference setting is a bug, it's odd that it only happens once in a while and not every time.  I'll try and see if I can reproduce.

**Comment by manongjohn** [comment]
This isn't a bug.

If you press Ctrl **before** you make your cell selections, it will also select keys and will move them when you move the cells

Sometimes you have to press the Ctrl/Shift/Alts in the xsheet before you make selections in order for it to work as desired.  If you made a selection and then tried to hit Ctrl/Shift/Alt afterwards, the effect you are going for may not always work.

**Comment by manongjohn** [comment]
However, I just discovered a bug with this feature due to the flip.  Can't always move the selection into an empty area within the same column, where as you could before.  Will find out what's wrong and fix.

**Comment by DarrenTAnims** [comment]
I can't move frames and keys at the same time. For me, this is the only behaviour I want. I NEVER want to move a drawing that has a key on it and NOT move the key. And I don't think other people do.
Also, much of the time when I really need to move keys with frames is when I want to extend a sequence. So I click and drag across multiple columns and want to drag them (and all the following frames and keys) down a number of frames, so I can insert a pause or extra animation. This is such a painful thing to do when you have keys set up at different places on multiple columns.

**Comment by beeheemooth** [comment]
@DarrenTAnims You talking about this?
![ezgif-1-649b5baca8](https://user-images.githubusercontent.com/25724133/45828127-80999a80-bd00-11e8-9d05-0c2eeb34e178.gif)

**Comment by DarrenTAnims** [comment]
It is, but on my system, the keys don't move! And I've got the Cell Dragging option to Cells and Column data (shown at the end of the clip)

![screencapture](https://user-images.githubusercontent.com/28575907/45828593-d108fc00-bcf0-11e8-
9fa0-1cd67cdd7184.gif)

**Comment by DarrenTAnims** [comment]
Right. Now I understand that you have to press CTRL BEFORE selecting. I can get this to work. But if I've set the option in prefs, it should move them without pressing CTRL.

Also, the CTRL option only works on frames you've drawn on. So, for instance, at the end of your clip, when the drawings have been moved to 13, you can click and drag across all columns at frame #10, then a drag handle appears. If you click and drag that down, it'll move all of the drawn frames below it down, but not the keys. I think it should move both with that option set, too.

**Comment by beeheemooth** [comment]
@DarrenTAnims Your video does not work for some reason.

It seems to me that the options "Cells and Column data" or "Cell Only" have nothing to do with moving frames (I do not even know what they are for)

![ezgif-1-f02a6adc54](https://user-images.githubusercontent.com/25724133/45829296-4d0c3f80-bd03-11e8-9426-3d0b05daef3c.gif)

**Comment by beeheemooth** [comment]
@DarrenTAnims Use "Insert Frame" / "Delete Frame" function to move frames together with keys

![ezgif-1-69f38c36cd](https://user-images.githubusercontent.com/25724133/45829933-ac1e8400-bd04-11e8-8fed-e5484ac8f441.gif)

**Comment by DarrenTAnims** [comment]
That's a great tip. Exactly what I wanted. I've just remapped my keyboard for Shift+Insert to do so.

**Comment by snapai** [comment]
RAHBOOTS on the Discord has pointed out to us - "insert frames" always inserts frames on ALL unlocked columns, there doesn't seem to be a way to do this with just one column short of locking all others, and still having the keys move with the frames. 
![singlecolumn](https://user-images.githubusercontent.com/5098372/50402998-14985980-0769-11e9-82ef-43baa094588c.gif)

(also unrelated, any way to set github to allow mp4 or webm video here? That clip is 5MB but the mp4 was 500kb)

**Comment by gab3d** [comment]
I'll bring this back...

I think that either:
- The **Cell-dragging Behavior** option should also take care of moving keys along with cells *inside* their column.
  or
- There should be another specific option for always selecting keys along with columns.

...what is clear is that this is expected behavior by anyone that works with other major graphics software, and there is also currently no UI hint that lets people that face this issue to discover it "the easy way", so there are big chances they end up thinking OT doesn't allow for such a basic task. What of course is bad... 🤔

**Comment by Nikhil-V-maker** [comment]
In Windows, initially hold Ctrl and select all the keys which you want to move along with frames. Don't leave the Ctrl, now press the first or last frame among the frames which you want to move, now along with Ctrl press shift and now click either last or first frame respectively, you can see all the frames as well as keys selected. Now press Ctrl+x to cut or Ctrl+c and paste wherever you want.

**Comment by RodneyBaker** [comment]
As the original reported has ghosted themself this is a candidate for closer.
However, as the report does collect quiet a lot of information for now we'll move to the Opentoonz-Docs repo.
